pkg/log/io/io.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io
+
+import (
+	"log/slog"
+	"os"
+)
+
+// IOController manages the input and output for the CLI
+type IOController struct {
+	in     *InputController
+	out    *OutputController
+	logger *oktetoLogger
+}
+
+// NewIOController returns a new input/output controller
+func NewIOController() *IOController {
+	ioController := &IOController{
+		out:    newOutputController(os.Stdout),
+		in:     newInputController(os.Stdin),
+		logger: newOktetoLogger(),
+	}
+	return ioController
+}
+
+// In returns the input reader
+func (ioc *IOController) In() *InputController {
+	return ioc.in
+}
+
+// Out returns the output writer
+func (ioc *IOController) Out() *OutputController {
+	return ioc.out
+}
+
+// Logger returns the logger
```

Comment:
can we add some comment in regards of what is the difference between output and logger? 🤔  right now i dont have the difference clear

Reply:
Ok, I'll add it there but the main difference is that the log can be shown to the user only if the level the user has set is lower than the log. For example, if there is a `logger.Info("this is a log")` we'll only show to the user `this is a log` if the level of the logger is `info` or `debug`, but not if it's set to `warning`. By the other hand the output will be always shown to the user regarding of the log level

Reply:
Done at [afb3857](https://github.com/okteto/okteto/pull/4014/commits/afb3857cdc73a18f04fd04643bc581e1da09ffb3)